composables/useVerifyAccount.ts

```diff
@@ -18,20 +19,34 @@ export default function useVerifyAccount() {
   const walletStore = useWalletStore()
   const { accountId } = useAuth()
   const signedMessage = computed(() => walletStore.getSignedMessage)
+  const { config: wagmiConfig } = useWagmi()
+
+  const signMessageEvm = async (address, message) => {
+    const signedMessage = await _signMessageEvm(wagmiConfig, {
+      account: address,
+      message: message,
+    })
+
+    console.log('Signed evm message:', { message, address, signedMessage })
+    return signedMessage
+  }
 
   const getSignedMessage = async () => {
-    if (!accountId.value || isEthereumAddress(accountId.value)) {
-      // TODO: handle EVM address verification
-      throw new Error('This address is not currently supported')
+    const isEvmAddress = isEthereumAddress(accountId.value)
+
+    if (!accountId.value) {
+      throw new Error('Please connect your wallet first')
     }
     if (signedMessage.value) {
       return signedMessage.value
     }
 
-    const signature = await signMessage(
-      toSubstrateAddress(accountId.value),
-      SIGNATURE_MESSAGE,
-    )
+    const signature = isEvmAddress
```

Comment:
```suggestion
    const signature = isEthereumAddress(accountId.value)
```

since `isEvmAddress` is not used elsewhere